packages/flutter/lib/src/painting/text_span.dart

```diff
@@ -80,11 +94,75 @@ class TextSpan {
 
   /// A gesture recognizer that will receive events that hit this text span.
```

Comment:
I mean, this is one of those cases where the API is bad.  I'm not sure what would be involved in making a better API.